### PR TITLE
Add changelog for dune 3.9.0

### PR DESCRIPTION
**Pull request by emillon**

### CODE REVIEW

**Comment by emillon** [comment]
cc @tmattio

**Comment by tmattio** [comment]
Thanks @emillon! I've pushed some minor improvements, let me know if the test looks good to you and it's ready to merge.

File: data/changelog/dune/2023-06-30-dune-3.9.0.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  
+  - Allow `(include_subdirs qualified)` to be used when libraries define a
+    `(modules ...)` field ([#7797](https://github.com/ocaml/dune/pull/7797), fixes [#7597](https://github.com/ocaml/dune/pull/7597), @anmonteiro)
+  
+  - Allow multiple globs in library's `(stdlib (internal_modules ..))`
+    (@anmonteiro, [#7878](https://github.com/ocaml/dune/pull/7878))
+  
+  ## Changed
+  
+  - Do not re-run OCaml syntax files on every iteration of the watch mode. This
```
Comment (christinerose) [inline data/changelog/dune/2023-06-30-dune-3.9.0.md]:
```suggestion
  - Do not rerun OCaml syntax files on every iteration of the watch mode. This
```

**Review by christinerose — APPROVED** [review]

**Comment by emillon** [comment]
Sure!

**Pull request merged**